src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -158,23 +159,34 @@ class SidebarLinks extends React.Component {
         this.state = {
             currentlyViewedReportID: props.currentlyViewedReportID,
             orderedReports: [],
+            priorityMode: props.priorityMode,
             unreadReports: SidebarLinks.getUnreadReports(props.reports || {}),
         };
     }
 
     static getDerivedStateFromProps(nextProps, prevState) {
         const shouldReorder = SidebarLinks.shouldReorder(nextProps, prevState.orderedReports, prevState.currentlyViewedReportID, prevState.unreadReports);
+        const switchingPriorityModes = nextProps.priorityMode !== prevState.priorityMode;
+
+        // Pull the reports we want to show on the left hand nav
         const recentReports = SidebarLinks.getRecentReports(nextProps);
-        const orderedReports = shouldReorder
+
+        // Determine whether we want to re-order/sort the reports on the left hand nav
+        const orderedReports = shouldReorder || switchingPriorityModes
             ? recentReports
-            : _.map(prevState.orderedReports,
-                orderedReport => _.chain(recentReports)
-                    .filter(recentReport => orderedReport.reportID === recentReport.reportID)
-                    .first()
-                    .value());
+            : _.chain(prevState.orderedReports)
+
+            // To preserve the order of the conversations, we map over the previous state's order of reports.
```

Comment:
I probably have to study the changes in `SidebarLinks` some more, but any ideas why it is necessary to "preserve the order" of these conversations?

Reply:
I think we didn't want the lefthand nav to reorder themselves too frequently. Specifically when a user begins to type a message to a user (who might be halfway down the list). It won't immediately pop it to them to the top of the list, unless they leave a draft for that user. 

Honestly, I'm not entirely sure. I saw that you had authored a few thing in this file so I was hoping you'd have a strong idea of why haha .

Original Issue for lefthand re-order here: https://github.com/Expensify/App/issues/5166 with the PR here https://github.com/Expensify/App/pull/5317

Reply:
Yeah, I did set up some things in the early days, but barely recognize it now.